src/__tests__/Migration.test.ts

```diff
@@ -28,11 +43,28 @@ afterAll(async () => {
 
 test('runs a migration correctly edit database', async () => {
   await db.connection('mongo').transaction(async (session) => {
-    const migration = new TestMigration('mongo', logger, session);
+    const migration = new TestMigration('mongo', logger, session, 'migration');
     await migration.execUp();
   });
   const collections = await (await db.connection('mongo').database())
     .listCollections()
     .toArray();
   expect(collections).toHaveLength(1);
 });
+
+test('do not revert all migrations if one fails', async () => {
+  await db.connection('mongo').transaction(async (session) => {
+    const migration = new TestMigration('mongo', logger, session, 'migration2');
+    const failMigration = new FailTestMigration('mongo', logger, session);
+
+    await migration.execUp();
+    const t = async () => {
+      await failMigration.execUp();
+    };
+    await expect(t).rejects.toStrictEqual(new Error('boom'));
+  });
+  const collections = await (await db.connection('mongo').database())
+    .listCollections()
+    .toArray();
+  expect(collections).toHaveLength(2);
```

Comment:
This depends on the correct execution of the previous test no?

Shouldn't you rather test the presence of the `migration2` collection to make this test isolated?

Reply:
I was adressing the wrong issue.